### PR TITLE
Test taskloop strict grainsize

### PR DESCRIPTION
**Pull request by krishols**
grainsize(strict:) controls logical iterations performed per task and I am unsure how to test this. Only have the skeleton of the test for now.

### CODE REVIEW

File: tests/5.1/strict/test_strict_grainsize.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  int scalar_var = 6;
+  int arr[N];
+  int newarr[N];
+  int sum =0;
+  for (int i=0; i<N; i++){
+        arr[i] = i;
+        sum += arr[i];
+ }
+#pragma omp parallel shared(arr, newarr)
+  {
```
Comment (spophale) [inline tests/5.1/strict/test_strict_grainsize.c:32]: #pragma amp single

**Comment by krishols** [comment]
Updated test - grainsize(strict) is not currently implemented but test passes without the strict modifier for now.

File: tests/5.1/strict/test_strict_grainsize.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  int parallel_sum = 0; 
+  for (int i=0; i<N; i++){
+        arr[i] = 1;
+        sum += arr[i];
+ }
+#pragma omp parallel  
+#pragma omp single
+#pragma omp taskloop grainsize(strict:1000)
+  for (int i = 0; i < N; i++) {
+  	parallel_sum += arr[i];
```
Comment (tob2) [inline tests/5.1/strict/test_strict_grainsize.c:32]: This works with GCC 12 – but only if I add `reduction(+:parallel_sum)`

Comment (krishols) [reply tests/5.1/strict/test_strict_grainsize.c]: Good catch - thanks! Updated.

**Review by tmh97 — APPROVED** [review]
good-to-go runs on summit w/ latest llvm(clang) when i remove the new feature

**Review by mjcarr458 — APPROVED** [review]
Looks good

**Pull request merged**